## Import the data

In [1]:
import pandas as pd
import regex as re
from bs4 import BeautifulSoup as bs
import os
import time

### Option 1: Beautiful Soup (for TEI)

I helped myself with the tutorial at https://komax.github.io/blog/text/python/xml/parsing_tei_xml_python/

To get correct text (with punctuation marks separated from the words), we need to extract them from the annotated XLM TEI file.

In [7]:
# Get a list of TEI files

path = "ParlaMint-SI.TEI.ana"
dir_list = os.listdir(path)

# Keep only files with parliamentary sessions:

parl_list = []

for i in dir_list:
	if "ParlaMint-SI_" in i:
		parl_list.append(i)

len(parl_list)


414

The number of all files is 414 sessions.

In [49]:
# Create an empty df
df = pd.DataFrame({"file": [""], "sentence_id": [""], "text": [""], "proper_nouns": [""]})

In [30]:
# Create a regex to separate word id into segment id and word index
word_seg_re = re.compile("(.*)\.(\d+)")

In [50]:
for doc in parl_list:

	# Open the file
	file = open(f"ParlaMint-SI.TEI.ana/{doc}", "r")
	content = bs(file, 'lxml')

	# Create a list of sentences
	sentence_list = []
	sen_id_list = []
	proper_nouns_list = []

	# Search for all segments
	seg_list = content.find_all("seg")

	for segment in seg_list:
		sentences = segment.find_all("s")
		for sentence in sentences:
			# Get text, replace \n with a space and remove spaces from the beginning and end of string
			sentence_list.append(sentence.getText().replace("\n", " ").strip(" "))
			
			# Add information on sentence id
			sen_id = sentence["xml:id"]
			sen_id_list.append(sen_id)

			# Add information on proper nouns
			current_proper_nouns_list = []

			result = sentence.find_all("name", type = "PER", recursive=False)

			if len(result) > 0:
				word_dict = {}
				for i in result:
					words = i.find_all("w", recursive = False)
					for word in words:
						current_name = word.getText()
						current_lemma = word["lemma"]
						current_word_id = word["xml:id"]
						current_word_index = word_seg_re.findall(current_word_id)[0][1]
						# I'll substract one from the word index, because indexing in the TEI file starts with 1, not 0
						current_word_index = int(current_word_index) - 1
						word_dict[current_word_index] = [current_name, current_lemma]
				
				current_proper_nouns_list.append(word_dict)

			proper_nouns_list.append(current_proper_nouns_list)

	new_df = pd.DataFrame({"sentence_id": sen_id_list, "text": sentence_list, "proper_nouns": proper_nouns_list})
	new_df["file"] = doc

	# Merge df to the previous df
	df = pd.concat([df, new_df])

In [51]:
df.head()

file  \
0                                                  
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   
2  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   
3  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   

                                    sentence_id  \
0                                                 
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg1.1   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1   
2  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg3.1   
3  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg3.2   

                                                text  \
0                                                      
0  Spoštovane kolegice poslanke , spoštovani kole...   
1  Začenjam nadaljevanje 12. seje  Državnega zbor...   
2  Obveščen sem , da se današnje seje ne morejo u...   
3             Vsem prisotnim še enkrat lep pozdrav !   

                                        proper_nouns  
0                                                     
0                                                 []  
1                                                 []  
2  [{15: ['Eva', 'Eva'], 16: ['Irgl', 'Irgl'], 19...  
3                                                 []

In [52]:
df.describe(include="all")

file sentence_id  \
count                                         1078999     1078999   
unique                                            415     1078999   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.ana.xml               
freq                                             6186           1   

                text proper_nouns  
count        1078999      1078999  
unique        918207        46914  
top     Hvala lepa .           []  
freq           27651       978131

### Option 2: From CONLLU with regex

In [2]:
# Get a list of TEI files

path = "ParlaMint-SI.conllu"
dir_list = os.listdir(path)

# Keep only files with parliamentary sessions:

parl_list = []

# Filter out only relevant files
for i in dir_list:
	if "ParlaMint-SI_" in i:
		if ".conllu" in i:
			parl_list.append(i)

len(parl_list)


414

In [3]:
# Create an empty df
df = pd.DataFrame({"file": [""], "sentence_id": [""], "text": [""]})

In [18]:
# Create regexes
sen_re = re.compile("# sent_id = (.*?)\n")
text_re = re.compile("# text = (.*?)\n")

In [5]:
# Find all relevant information from the conllu files and save everything in a dataframe

for i in parl_list:

	file = open(f"ParlaMint-SI.conllu/{i}", "r").read()

	# Find sentence ids and texts
	sen_ids = sen_re.findall(file)
	texts = text_re.findall(file)
	
	new_df = pd.DataFrame({"sentence_id": sen_ids, "text": texts})
	new_df["file"] = i

	# Merge df to the previous df
	df = pd.concat([df, new_df])

df.head()

file  \
0                                                 
0  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
1  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
2  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
3  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   

                                    sentence_id  \
0                                                 
0  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg1.1   
1  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg2.1   
2  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3.1   
3  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3.2   

                                                text  
0                                                     
0           Nadaljujemo s prekinjeno 17. sejo zbora.  
1  Prehajamo na 2. TOČKO DNEVNEGA REDA, TO JE NA ...  
2  Predlog zakona je v obravnavo zboru predložil ...  
3  V zvezi s tem predlogom zakona Odbor za zdravs...

### Continue editing the dataframe (from option 1 or 2)

In [53]:
# Remove the first row
df = df.drop([0], axis="index")

# Reset index
df = df.reset_index(drop=True)

# Show the results
df.describe(include="all")

file  \
count                                         1078584   
unique                                            414   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.ana.xml   
freq                                             6185   

                                         sentence_id          text  \
count                                        1078584       1078584   
unique                                       1078584        918159   
top     ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1  Hvala lepa .   
freq                                               1         27651   

       proper_nouns  
count       1078584  
unique        46912  
top              []  
freq         977718

The Parlamint-SI consists of 414 files and 1,078,584 sentences.

In [54]:
# Add information on length
df["length"] = df["text"].str.split().str.len()

df.head(2)


file  \
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   

                                    sentence_id  \
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg3.1   

                                                text  \
0  Začenjam nadaljevanje 12. seje  Državnega zbor...   
1  Obveščen sem , da se današnje seje ne morejo u...   

                                        proper_nouns  length  
0                                                 []       7  
1  [{15: ['Eva', 'Eva'], 16: ['Irgl', 'Irgl'], 19...      89

In [57]:
df.length.describe().round()

count    1078584.0
mean          22.0
std           17.0
min            1.0
25%            9.0
50%           18.0
75%           30.0
max          532.0
Name: length, dtype: float64

In [63]:
df.tail()

file  \
1078579  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.ana.xml   
1078580  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.ana.xml   
1078581  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.ana.xml   
1078582  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.ana.xml   
1078583  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.ana.xml   

                                            sentence_id  \
1078579  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.seg540.1   
1078580  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.seg541.1   
1078581  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.seg541.2   
1078582  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.seg543.1   
1078583  ParlaMint-SI_2018-02-13-SDZ7-Redna-38.seg544.1   

                                                      text  \
1078579  Na glasovanje dajem naslednji sklep :  Državni...   
1078580                                        Glasujemo .   
1078581  Navzočih je 65 poslank in poslancev , za je gl...   
1078582                 Ugotavljam , da sklep ni sprejet .   
1078583  S tem zaključujem 1. točko dnevnega reda in pr...   

                                              proper_nouns  length  
1078579  [{24: ['Dejana', 'Dejan'], 25: ['Židana', 'Žid...      41  
1078580                                                 []       2  
1078581                                                 []      14  
1078582                                                 []       7  
1078583                                                 []      23

In [64]:
# Save the dataframe
df.to_csv("Parlamint-SI-sentences.csv", sep="\t")

## Machine translate a sample

In [2]:
# Open the file, created in the previous step
df = pd.read_csv("Parlamint-SI-sentences.csv", sep="\t", index_col=0)

df.head()


/home/tajak/anaconda3/lib/python3.9/site-packages/numpy/lib/arraysetops.py:583: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)


file  \
0  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
1  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
2  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
3  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   
4  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.conllu   

                                    sentence_id  \
0  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg2.1   
1  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3.1   
2  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3.2   
3  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg4.1   
4  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg5.1   

                                                text  length  \
0  Prehajamo na 2. TOČKO DNEVNEGA REDA, TO JE NA ...      28   
1  Predlog zakona je v obravnavo zboru predložil ...       9   
2  V zvezi s tem predlogom zakona Odbor za zdravs...      22   
3  Besedo dajem predsednici odbora Anji Bah Žiber...      12   
4                          Hvala lepa, predsedujoči.       3   

                                       seg_id  
0  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg2  
1  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3  
2  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg3  
3  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg4  
4  ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg5

In [10]:
df.describe(include="all")

file  \
count                                        1078584   
unique                                           414   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.conllu   
freq                                            6185   
mean                                             NaN   
std                                              NaN   
min                                              NaN   
25%                                              NaN   
50%                                              NaN   
75%                                              NaN   
max                                              NaN   

                                         sentence_id         text  \
count                                        1078584      1078584   
unique                                       1078584       918195   
top     ParlaMint-SI_2020-05-27-SDZ8-Redna-17.seg2.1  Hvala lepa.   
freq                                               1        27653   
mean                                             NaN          NaN   
std                                              NaN          NaN   
min                                              NaN          NaN   
25%                                              NaN          NaN   
50%                                              NaN          NaN   
75%                                              NaN          NaN   
max                                              NaN          NaN   

              length                                       seg_id  
count   1.078584e+06                                      1078584  
unique           NaN                                       280629  
top              NaN  ParlaMint-SI_2014-11-20-SDZ7-Redna-02.seg69  
freq             NaN                                          147  
mean    1.875484e+01                                          NaN  
std     1.489782e+01                                          NaN  
min     1.000000e+00                                          NaN  
25%     7.000000e+00                                          NaN  
50%     1.600000e+01                                          NaN  
75%     2.600000e+01                                          NaN  
max     4.700000e+02                                          NaN

In [65]:
sample_files = list(df["file"].unique())[:3]
sample_files

['ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml',
 'ParlaMint-SI_2019-12-19-SDZ8-Redna-14.ana.xml',
 'ParlaMint-SI_2019-03-27-SDZ8-Redna-06.ana.xml']

In [66]:
# Take the first 3 files for a sample

df_sample = df[df.file.isin(sample_files)]
df_sample.describe(include="all")

file  \
count                                           11750   
unique                                              3   
top     ParlaMint-SI_2019-12-19-SDZ8-Redna-14.ana.xml   
freq                                             5958   
mean                                              NaN   
std                                               NaN   
min                                               NaN   
25%                                               NaN   
50%                                               NaN   
75%                                               NaN   
max                                               NaN   

                                         sentence_id          text  \
count                                          11750         11750   
unique                                         11750         10639   
top     ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1  Hvala lepa .   
freq                                               1           312   
mean                                             NaN           NaN   
std                                              NaN           NaN   
min                                              NaN           NaN   
25%                                              NaN           NaN   
50%                                              NaN           NaN   
75%                                              NaN           NaN   
max                                              NaN           NaN   

       proper_nouns        length  
count         11750  11750.000000  
unique         1052           NaN  
top              []           NaN  
freq          10499           NaN  
mean            NaN     20.336936  
std             NaN     15.324123  
min             NaN      1.000000  
25%             NaN      9.000000  
50%             NaN     17.000000  
75%             NaN     28.000000  
max             NaN    185.000000

The sample file has 11750 sentences, out of which 1251 (11%) have proper nouns.

In [69]:
# Save sample file
df_sample.to_csv("Parlamint-SI-sample.csv")

In [2]:
# Open the entire file
df = pd.read_csv("Parlamint-SI-sentences.csv", sep="\t", index_col = 0)
df.describe(include="all")

/home/tajak/anaconda3/lib/python3.9/site-packages/numpy/lib/arraysetops.py:583: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)


file  \
count                                         1078584   
unique                                            414   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.ana.xml   
freq                                             6185   
mean                                              NaN   
std                                               NaN   
min                                               NaN   
25%                                               NaN   
50%                                               NaN   
75%                                               NaN   
max                                               NaN   

                                         sentence_id          text  \
count                                        1078584       1078584   
unique                                       1078584        918159   
top     ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1  Hvala lepa .   
freq                                               1         27651   
mean                                             NaN           NaN   
std                                              NaN           NaN   
min                                              NaN           NaN   
25%                                              NaN           NaN   
50%                                              NaN           NaN   
75%                                              NaN           NaN   
max                                              NaN           NaN   

       proper_nouns        length  
count       1078584  1.078584e+06  
unique        46912           NaN  
top              []           NaN  
freq         977718           NaN  
mean            NaN  2.180275e+01  
std             NaN  1.691792e+01  
min             NaN  1.000000e+00  
25%             NaN  9.000000e+00  
50%             NaN  1.800000e+01  
75%             NaN  3.000000e+01  
max             NaN  5.320000e+02

The entire ParlaMint has 1,078,584 sentences and 23,516,095 words which would take approximately 10 hours to translate.

Translate additional files to have a larger sample - let's say 37 more files so that we have 10% of files translated.

In [9]:
sample_files = list(df["file"].unique())[3:40]

df_sample2 = df[df.file.isin(sample_files)]
df_sample2.describe(include="all")


file  \
count                                          104243   
unique                                             37   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.ana.xml   
freq                                             6185   
mean                                              NaN   
std                                               NaN   
min                                               NaN   
25%                                               NaN   
50%                                               NaN   
75%                                               NaN   
max                                               NaN   

                                           sentence_id          text  \
count                                           104243        104243   
unique                                          104243         92471   
top     ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg2.1  Hvala lepa .   
freq                                                 1          2717   
mean                                               NaN           NaN   
std                                                NaN           NaN   
min                                                NaN           NaN   
25%                                                NaN           NaN   
50%                                                NaN           NaN   
75%                                                NaN           NaN   
max                                                NaN           NaN   

       proper_nouns         length  
count        104243  104243.000000  
unique         6536            NaN  
top              []            NaN  
freq          95022            NaN  
mean            NaN      21.660764  
std             NaN      16.825063  
min             NaN       1.000000  
25%             NaN       9.000000  
50%             NaN      18.000000  
75%             NaN      30.000000  
max             NaN     532.000000

The additional sample has 104,243 sentences and 2,257,983 words. The translation should take a bit less than an hour.

### Machine translate

In [ ]:
# Install easynmt
#!pip install -q -U easynmt

In [13]:
from easynmt import EasyNMT

# Define the model - The system will automatically detect the suitable Opus-MT model and load it.
model = EasyNMT('opus-mt')

In [14]:
# Create a list of sentences from the df
sentence_list = df_sample2.text.to_list()
len(sentence_list)

104243

In [16]:
#Translate the list of sentences - you need to provide the source language as it is in the name of the model (zls - South Slavic)
translation_list = model.translate(sentence_list, source_lang = 'zls', target_lang='en')

# Add the translations to the df
df_sample2["translation"] = translation_list

df_sample2.head()

/tmp/ipykernel_12711/3063041765.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_sample2["translation"] = translation_list


file  \
11750  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.ana.xml   
11751  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.ana.xml   
11752  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.ana.xml   
11753  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.ana.xml   
11754  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.ana.xml   

                                          sentence_id  \
11750  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg2.1   
11751  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg3.1   
11752  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg4.1   
11753  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg4.2   
11754  ParlaMint-SI_2016-11-07-SDZ7-Izredna-36.seg5.1   

                                                    text  \
11750  Začenjam 36. izredno sejo  Državnega zbora  , ...   
11751  Obveščen sem , da se današnje seje ne morejo u...   
11752  Na sejo sem vabil predsednika  Vlade  dr.  Mir...   
11753                    Vse prisotne lepo pozdravljam !   
11754  Prehajamo na določitev dnevnega reda 36. izred...   

                                            proper_nouns  length  \
11750                                                 []      26   
11751  [{16: ['Bojana', 'Bojana'], 17: ['Muršič', 'Mu...     114   
11752  [{7: ['Mira', 'Mir'], 8: ['Cerarja', 'Cerar'],...      26   
11753                                                 []       5   
11754                                                 []      11   

                                             translation  
11750  I shall begin the 36th extraordinary meeting o...  
11751  I am informed that today's meeting cannot be a...  
11752  At the meeting, I invited the Prime Minister D...  
11753                       I welcome all those present!  
11754  We move to the agenda of the 36th extraordinar...

The first sample which consists of 11750 sentences and 238,959 (3 files) was translated in 6 minutes. The second sample (104,243 sentences, 2,257,983 words, 37 files) was translated in 52 minutes.

In [18]:
# Open the first sample and merge the second sample with it
df_sample = pd.read_csv("Parlamint-SI-sample-translated.csv", sep="\t", index_col = 0)

df_sample_merged = pd.concat([df_sample, df_sample2])

df_sample_merged = df_sample_merged.reset_index(drop = True)

df_sample_merged.describe(include="all")

file  \
count                                          115993   
unique                                             40   
top     ParlaMint-SI_2016-11-23-SDZ7-Redna-24.ana.xml   
freq                                             6185   
mean                                              NaN   
std                                               NaN   
min                                               NaN   
25%                                               NaN   
50%                                               NaN   
75%                                               NaN   
max                                               NaN   

                                         sentence_id          text  \
count                                         115993        115993   
unique                                        115993        102752   
top     ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1  Hvala lepa .   
freq                                               1          3029   
mean                                             NaN           NaN   
std                                              NaN           NaN   
min                                              NaN           NaN   
25%                                              NaN           NaN   
50%                                              NaN           NaN   
75%                                              NaN           NaN   
max                                              NaN           NaN   

       proper_nouns         length           translation  
count        115993  115993.000000                115993  
unique         7333            NaN                101985  
top              []            NaN  Thank you very much.  
freq         105521            NaN                  3169  
mean            NaN      21.526661                   NaN  
std             NaN      16.683888                   NaN  
min             NaN       1.000000                   NaN  
25%             NaN       9.000000                   NaN  
50%             NaN      18.000000                   NaN  
75%             NaN      30.000000                   NaN  
max             NaN     532.000000                   NaN

The extended sample has 40 files, 115,993 sentences and 2,496,942 words. It represents rougly 10% of the entire ParlaMint. In the sample, around 10% of sentences consist of proper names.

In [19]:
df_sample_merged.head(2)

file  \
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.ana.xml   

                                    sentence_id  \
0  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg2.1   
1  ParlaMint-SI_2019-10-23-SDZ8-Redna-12.seg3.1   

                                                text  \
0  Začenjam nadaljevanje 12. seje  Državnega zbor...   
1  Obveščen sem , da se današnje seje ne morejo u...   

                                        proper_nouns  length  \
0                                                 []       7   
1  [{15: ['Eva', 'Eva'], 16: ['Irgl', 'Irgl'], 19...      89   

                                         translation  
0  I am starting a continuation of the 12th meeti...  
1  I am informed that today's meeting cannot be a...

In [20]:
df_sample_merged.tail(2)

file  \
115991  ParlaMint-SI_2020-04-30-SDZ8-Izredna-36.ana.xml   
115992  ParlaMint-SI_2020-04-30-SDZ8-Izredna-36.ana.xml   

                                             sentence_id  \
115991  ParlaMint-SI_2020-04-30-SDZ8-Izredna-36.seg218.1   
115992  ParlaMint-SI_2020-04-30-SDZ8-Izredna-36.seg219.1   

                                                     text proper_nouns  \
115991                 Ugotavljam , da je sklep sprejet .           []   
115992  S tem zaključujem to točko in 36. izredno sejo...           []   

        length                                        translation  
115991       7          I see that the decision has been adopted.  
115992      11  I hereby conclude this point and the 36th extr...

In [25]:
# Save the df
df_sample_merged.to_csv("Parlamint-SI-bigger-sample-translated.csv", sep = "\t")